src/peep/peep.c

```diff
@@ -9021,12 +9021,12 @@ money32 set_peep_name(int flags, int state, uint16 sprite_index, uint8* text_1,
 	}
 
 	utf8* fullText = RCT2_ADDRESS(0x00F1AEF6, utf8);
-	if (flags & GAME_COMMAND_FLAG_APPLY) {
+	//if (flags & GAME_COMMAND_FLAG_APPLY) { // this check seems to be useless and causes problems in multiplayer
```

Comment:
I don't think so, fullText is just a temporary static used to read in the name over multiple calls to set the name.  You will notice a few lines down, it will check again if the GAME_COMMAND_FLAG_APPLY is not set and will free the string and return, not doing anything.  In fact the first call to check if it is possible to rename will not work without uncommenting this because the temporary string is not even set, which was the problem in multiplayer.  Probably an original bug that only occurs in multiplayer because in singleplayer this is called in the order of: (where the number is state and a is if the FLAG_APPLY is set)
1,1a,2,2a,0,0a
but in multiplayer because of delay:
1,2,0,1a,2a,0a